### ISSUE TITLE
Minibatch training for parametric umap

### ISSUE THREAD

**Issue by theahura** [lmcinnes/umap#925]
I'm trying to train a parametric umap model on ~100M clip embeddings sized 512D. It's a bit too large to run the whole dataset in memory; does the parametric umap model support batch training?

Thanks!

**Comment by timsainb** [comment]
Yes, you can either load up part of the dataset and train, then load up another part and train on that, etc or you can write your own mini batch loader in tensorflow and train on that.
https://colab.research.google.com/drive/1WkXVZ5pnMrm17m0YgmtoNjM_XHdnE5Vp?usp=sharing

for the most part you’re going to get diminishing returns in terms of visual embedding quality when you add more data, and for most part embedding the biggest random sample of your data that will fit in memory will be just fine.
